src/basic/fs-util.c

```diff
@@ -116,7 +116,7 @@ int rename_noreplace(int olddirfd, const char *oldpath, int newdirfd, const char
 int readlinkat_malloc(int fd, const char *p, char **ret) {
         size_t l = PATH_MAX;
 
-        assert(p);
+        assert(fd >= 0 || p);
```

Comment:
AT_FDCWD should also be ok for fd, and is negative.

also maybe add a check `if (fd == AT_FDCWD && isempty(p)) return -EISDIR;`?

i.e. usually such a thing would mean current working dir after all, but that's definitely not a symlink, hence returning EISDIR makes sense